elyra/cli/pipeline_app.py

```diff
@@ -408,7 +434,124 @@ def describe(json_option, pipeline_path):
         click.echo(json.dumps(describe_dict, indent=indent_length))
 
 
+@click.command()
+@click.argument('pipeline_path',
+                type=Path,
+                callback=validate_pipeline_path)
+@click.option('--runtime-config',
+              required=True,
+              help='Runtime configuration name.')
+@click.option('--output',
+              required=False,
+              type=Path,
+              help='Exported file name (including optional path). Defaults to '
+                   ' the current directory and the pipeline name.')
+@click.option('--overwrite',
+              is_flag=True,
+              help='Overwrite output file if it already exists.')
+def export(pipeline_path, runtime_config, output, overwrite):
+    """
+    Export a pipeline to a runtime-specific format
+    """
+
+    click.echo()
+    print_banner("Elyra pipeline export")
+
+    runtime_schema = _get_runtime_schema_name(runtime_config)
+    runtime_type = _get_runtime_type(runtime_config)
+
+    pipeline_definition = \
+        _preprocess_pipeline(pipeline_path, runtime=runtime_schema, runtime_config=runtime_config)
+
+    # Verify that the pipeline's runtime type is compatible with the
+    # runtime configuration
+    pipeline_runtime_type = pipeline_definition.get('pipelines', [{}])[0]\
+                                               .get('app_data', {})\
+                                               .get('runtime_type', 'Generic')
+    if pipeline_runtime_type and\
+       pipeline_runtime_type != 'Generic' and\
+       pipeline_runtime_type != runtime_type:
+        raise click.BadParameter(f"The runtime configuration type '{runtime_type}' does not match "
+                                 f"the pipeline's runtime type '{pipeline_runtime_type}'.",
+                                 param_hint='--runtime-config')
+
+    supported_export_formats = {
+        'KUBEFLOW_PIPELINES': [
+            {
+                'format': 'yaml',
+                'file_suffixes': ['.yaml', '.yml'],
+                'default_file_suffix': '.yaml'
+            }
+        ],
+        'APACHE_AIRFLOW': [
+            {
+                'format': 'py',
+                'file_suffixes': ['.py'],
+                'default_file_suffix': '.py'
+            }
+        ]
+    }
```

Comment:
Should this information be part of [`RuntimeTypeResources`](https://github.com/elyra-ai/elyra/blob/deb51ab926acfd98cfe5696acc772640cb2b831a/elyra/pipeline/runtime_type.py#L61)? They already include an `export_file_types` property.

Reply:
Yes, great suggestion that eliminates duplication! I've changed the implementation accordingly.